deepspeed/runtime/utils.py

```diff
@@ -48,6 +48,28 @@ def __init__(self, params):
         self.param_groups.append({'params': params})
 
 
+graph_cache = {}
+
+
+def graph_process(replay_first_step, func, *args, **kwargs):
+    # `func` should only contain operations on the GPU
+    # Please ensure that the memory address of the data required by 'func' remains constant
```

Comment:
I have two questions here. 
1. How can this requirement of data memory address be guaranteed or verified? 
2. What happens if the memory address changes, will the kernels crash?

Reply:
@inkcherry, hope you did not miss my questions. Thanks!